File: JAVA/ATM_Design/src/main/java/org/ceo/services/DebitCardManagerService.java
package org.ceo.services;

import org.ceo.models.Card;

public class DebitCardManagerService implements CardManagerService{
    @Override
    public boolean validateCard(Card card, String pin) {
        return false;
    }

    @Override
    public boolean validateWithdrawal(int transactionId, double amount) {
        return false;
    }

    @Override
    public boolean doTransaction(Card card, double amount, int transactionId) {
        return false;
    }
}

File: C++/ATM-System/src/services/DebitCardManagerService.cpp
#include "../services/CardManagerService.h"

class DebitCardManagerService:public CardManagerService{
    public:
    bool validateCard(Card card,std::string pin) override{
        return false;
    }
    bool validateWithdrawal(int transactionId,double amount) override{
        return false;
    }
    bool doTransaction(Card card,double amount,int transactionId) override{
        return false;
    }
};